app/main.py

```diff
@@ -0,0 +1,30 @@
+import os
+import sys
+
+from flask import abort, Flask, jsonify, request
+
+from search.recommendation import SearchRecommendation
+
+import conf
+from memorize import CacheMissError
+
+
+sys.path.append(os.path.dirname(__file__))
+app = Flask(__name__)
+
+@app.route('/')
+def main():
+    q = request.args.get('q')
+    if not q:
+        abort(400)
+    try:
+        search = SearchRecommendation(q, request)
```

Comment:
would be good to add a TODO about checking the cache before kicking off the full search